crates/nu-protocol/src/eval_const.rs

```diff
@@ -185,24 +185,28 @@ pub(crate) fn create_nu_constant(engine_state: &EngineState, span: Span) -> Valu
         },
     );
 
+    // TODO: update docs
     // Create a system level directory for nushell scripts, modules, completions, etc
     // that can be changed by setting the NU_VENDOR_AUTOLOAD_DIR env var on any platform
     // before nushell is compiled OR if NU_VENDOR_AUTOLOAD_DIR is not set for non-windows
     // systems, the PREFIX env var can be set before compile and used as PREFIX/nushell/vendor/autoload
     record.push(
         "vendor-autoload-dir",
+        // TODO: update docs
         // pseudo code
         // if env var NU_VENDOR_AUTOLOAD_DIR is set, in any platform, use it
         // if not, if windows, use ALLUSERPROFILE\nushell\vendor\autoload
         // if not, if non-windows, if env var PREFIX is set, use PREFIX/share/nushell/vendor/autoload
         // if not, use the default /usr/share/nushell/vendor/autoload
 
         // check to see if NU_VENDOR_AUTOLOAD_DIR env var is set, if not, use the default
-        if let Some(path) = get_vendor_autoload_dir(engine_state) {
-            Value::string(path.to_string_lossy(), span)
-        } else {
-            Value::error(ShellError::ConfigDirNotFound { span: Some(span) }, span)
-        },
+        Value::list(
```

Comment:
I'm not sure I like this because we don't know what order things get sourced/used.

Reply:
We don't? Aren't lists ordered? Maybe this is just something to extend documentation on?

Reply:
Not that I know of. Doesn't that imply some internal sorting somewhere in nushell? If so, I'm pretty sure that doesn't happen.

Reply:
Why would this be unordered? `Value::List` just wraps an (ordered) `Vec`. The return value from `get_vendor_autoload_dir` is also sorted the same as the values in the original environment variable, as expected.

Reply:
I'm not sure I understand the problem here either - lists are ordered, we should parse `XDG_DATA_DIRS` in order.

Should it be expected that things on the left side of the list will load first or last? If the default is `/usr/local/share:/usr/share`, that kind of implies that the beginning of the list should load last (you'd want `/usr/local` stuff to be able to replace `/usr` stuff.)

Reply:
Yep, it should load right to left, so that the precedence is left to right. I still need to reverse the loading order for that to actually happen.